typer/main.py

```diff
@@ -631,9 +631,11 @@ def convertor(value: Any) -> Any:
 
 
 def generate_list_convertor(
-    convertor: Optional[Callable[[Any], Any]]
-) -> Callable[[Sequence[Any]], List[Any]]:
-    def internal_convertor(value: Sequence[Any]) -> List[Any]:
+    convertor: Optional[Callable[[Any], Any]], default_value: Optional[Any]
+) -> Callable[[Sequence[Any]], Optional[List[Any]]]:
+    def internal_convertor(value: Sequence[Any]) -> Optional[List[Any]]:
+        if default_value is None and len(value) == 0:
```

Comment:
bug. 

If `default_value is None`, then the second condition, which attempts to execute `len(None)`, will always result in an error."